src/main/scala/com/auth/example/HttpApp.scala

```diff
@@ -15,20 +10,23 @@ import org.http4s.server.Server
 import smithy4s.http4s.SimpleRestJsonBuilder
 import zio.interop.catz._
 import zio.{Fiber, FiberRef, Task, ULayer, ZLayer}
+import com.auth.example.domain.AuthedUser
 
 object HttpApp {
 
-  val resourceRoutes = FiberRef.make(Option.empty[User]).flatMap { userFRef =>
-    val authenticationService    = new AuthenticationServiceImpl(UserManagementRepository)
-    val authenticationMiddleware = new AuthenticationMiddlewareImpl(authenticationService, userFRef)
-    val authMiddleware           = new AuthMiddleware(authenticationMiddleware)
+  val resourceRoutes = FiberRef.make(Option.empty[AuthedUser]).flatMap { userRef =>
+    val authenticationService = new AuthenticationServiceImpl(UserManagementRepository)
+    val storeCurrentUser      = (user: AuthedUser) => userRef.set(Some(user))
+    val getCurrentUser        = userRef.get.someOrFail(BadRequest())
```

Comment:
I'm only passing functions to the relevant constructs: the middleware only needs to worry about storing the authenticated user once it's done its work. 

The service needs only to worry about accessing the authenticated user.